valohai/internals/download.py

```diff
@@ -1,8 +1,39 @@
 import os
 import tempfile
+from requests import Response
+from valohai.internals.utils import uri_to_filename, check_sha256_hashing
 
 
 # TODO: This is close to valohai-local-run. Possibility to merge.
+def resolve_datum(datum_id: str) -> Response:
+    try:
+        from valohai_cli.api import request  # type: ignore
+    except ImportError as ie:
+        raise RuntimeError("Can't resolve datum without valohai-cli") from ie
+    resp: Response = request(url=f"/api/v0/data/{datum_id}", method="GET", stream=True)
+    resp.raise_for_status()
+    return resp
+
+
+def verify_datum(response: Response, input_folder_path: str) -> str:
+    datum_obj = response.json()
+    filename = datum_obj["name"]
+    checksums = {
+        "md5": datum_obj["md5"],
+        "sha1": datum_obj["sha1"],
+        "sha256": datum_obj["sha256"],
+    }
+    file_path = os.path.join(input_folder_path, filename)
+    if filename in os.listdir(input_folder_path) and checksums[
+        "sha256"
+    ] == check_sha256_hashing(file_path):
```

Comment:
You can use `os.path.isfile` instead of enumerating the entire directory (which could be 10,000 files :) ) – and with the above suggestion of getting rid of `checksums`...
```suggestion
    if os.path.isfile(file_path) and check_sha256_hashing(file_path) == datum_obj["sha256"]:
```